Add whitenoise to serve static files

karimkawambwa opened the pull request:
## Description

The changes make whitenoise collect and serve the files in production.

**NOTE: This change was in production but not tracked.**

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)


## Screenshots


## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

kilemensi commented on Dockerfile:
```diff
@@ -33,7 +33,7 @@ ADD . /opt/code
 
 WORKDIR /opt/code/feinstaub
-USER root
+USER uid1000
```
Any advantage of running as non-root _in a container_ env?

karimkawambwa replied:
Originally feinstaub did this separation. I am just returning to the original behavour. I think they did this to separate access to log and static files. I can see most of them live in the uid1000 directory which is owned by uid1000.

karimkawambwa commented:
Removed the `postgresql-server-dev-all` install which was breaking travis tests due to dependencies fail. From exploring its use, it seems not important so I removed it to allow travis to proceed.

karimkawambwa commented:
> 2\. Why are we adding and where are we using `django.middleware.security.SecurityMiddleware`.

This is done to get security to pages given free by the SecurityMiddleware

kilemensi commented:
> > 2. Why are we adding and where are we using `django.middleware.security.SecurityMiddleware`.
> 
> This is done to get security to pages given free by the SecurityMiddleware

Cool, but I think you **_have_** to set them on `settings.py`. Is there any security feature that is enabled by just add  `django.middleware.security.SecurityMiddleware` to the `MIDDLEWARE`?

karimkawambwa commented:
> Cool, but I think you **_have_** to set them on `settings.py`. Is there any security feature that is enabled by just add `django.middleware.security.SecurityMiddleware` to the `MIDDLEWARE`?

@kilemensi thats true, i'll remove unused. am not sure if we should use it but i'll reserve that for another PR.
